.github/workflows/continuous-integration.yml

```
@@ -95,6 +129,8 @@ jobs:
   compile_frontend:
     name: Compile kadai-web
     runs-on: ubuntu-22.04
+    needs: [ detect_changes ]
+    if: needs.detect_changes.outputs.backend_changed == 'true' || needs.detect_changes.outputs.frontend_changed == 'true'
```

Why check for backend changed here?

```suggestion
    if: needs.detect_changes.outputs.frontend_changed == 'true'
```

---

We need to compile both frontend and backend regardless of which one changed. Otherwise, jobs:  release_artifacts , deploy_to_azure , and upload_to_sonar will fail because they always expect both frontend and backend builds to be available.